src/main.ts

```diff
@@ -1,16 +1,299 @@
+          issueNumber
+        }
+      }
+      break
+    }
+    case 'issues': {
+      if (github.context.payload.issue) {
+        ctx = {
+          body: github.context.payload.issue.body || '',
+          html_url: github.context.payload.issue.html_url || '',
+          source_id: github.context.payload.issue.id,
+          issueNumber
+        }
+      }
+      break
+    }
+    default:
+      core.info(`Unknown event: ${github.context.eventName}`)
+  }
+  return ctx
+}
+
+async function markCommentAsUpdating(
+  repo: string[],
+  toUpdateComment: number
+): Promise<void> {
+  try {
+    const comment = await octokit.rest.issues.getComment({
+      owner: repo[0],
+      repo: repo[1],
+      comment_id: toUpdateComment
+    })
+
+    const body = `Updating snippets in progress ...\n${comment.data.body || ''}`
+    await octokit.rest.issues.updateComment({
+      owner: repo[0],
+      repo: repo[1],
+      comment_id: toUpdateComment,
+      body
+    })
+  } catch (error) {
+    return
+  }
+}
+
+async function findGeneratedCommentByBotId(
+  repo: string[],
+  ghContext: GithubContext
+): Promise<number> {
+  const issueComments = await octokit.rest.issues.listComments({
+    owner: repo[0],
+    repo: repo[1],
+    issue_number: ghContext.issueNumber
+  })
+
+  const commentIdRegex = new RegExp('(id):([0-9]+)')
+
+  const commentsId: Map<string, number> = new Map(
+    issueComments.data.map(comment => {
+      const body = comment.body || ''
+      const match = commentIdRegex.exec(body)
+      let id = ''
+      if (match != null) {
+        id = match[2]
+      }
+      return [id, comment.id]
+    })
+  )
+
+  const toUpdateCommentId = commentsId.get(ghContext.source_id.toString()) || 0
+  return toUpdateCommentId
+}
+
+async function createComment(
+  comment: string,
+  repo: string[],
+  ghContext: GithubContext,
+  toUpdateCommentId: number
+): Promise<void> {
+  if (toUpdateCommentId !== 0) {
+    await octokit.rest.issues.updateComment({
+      owner: repo[0],
+      repo: repo[1],
+      comment_id: toUpdateCommentId,
+      body: comment
+    })
+  } else {
+    await octokit.rest.issues.createComment({
+      owner: repo[0],
+      repo: repo[1],
+      issue_number: ghContext.issueNumber,
+      body: comment
+    })
+  }
+}
+
+function generateCommentHeader(ghContext: GithubContext): string {
+  const header = `<!--\n${headerInfo}\nid:${ghContext.source_id}\n-->\n`
+  return header
+}
+
+function isAllowedEvent(): boolean {
+  return (
+    github.context.eventName === 'issue_comment' ||
+    github.context.eventName === 'issues'
+  )
+}
+
+function isCreatedByBot(): Boolean {
+  if (github.context.payload.comment) {
+    const body: string = github.context.payload.comment.body
+    return body.includes(headerInfo)
+  } else {
+    return false
+  }
+}
+
+function getSnippet(ghContext: GithubContext): Snippet[] {
+  const snippets: Snippet[] = []
+
+  const regex = new RegExp('```.*?(scala-cli(.*)([^]*?))```', 'g')
```

Comment:
I may also be mistaken with the above, since I don't really understand what is going on with the third group `([^]*?)`, shouldn't it be `(^.*?)`? Won't `[^]` literally match `^` char? Yet it all seems to be working. So I may be completely wrong, if so then disregard what I said.

Reply:
Nevermind, it seems that `.` does not include line terminators